server/controllers/manage/placementController.ts

```diff
@@ -1,24 +1,74 @@
 import type { Request, RequestHandler, Response } from 'express'
-import { PremisesService } from '../../services'
+import type {
+  ApprovedPremisesApplication,
+  ApprovedPremisesAssessment,
+  PlacementRequestDetail,
+  TimelineEvent,
+} from '@approved-premises/api'
+import {
+  ApplicationService,
+  AssessmentService,
+  PlacementRequestService,
+  PlacementService,
+  PremisesService,
+} from '../../services'
 
 import { DateFormats } from '../../utils/dateUtils'
-import { getBackLink } from '../../utils/placements'
+import { PlacementTab, getBackLink, placementTabItems } from '../../utils/placements'
+import { mapApplicationTimelineEventsForUi } from '../../utils/applications/utils'
 
 export default class PlacementController {
-  constructor(private readonly premisesService: PremisesService) {}
+  constructor(
+    private readonly applicationService: ApplicationService,
+    private readonly assessmentService: AssessmentService,
+    private readonly placementRequestService: PlacementRequestService,
+    private readonly placementService: PlacementService,
+    private readonly premisesService: PremisesService,
+  ) {}
 
   show(): RequestHandler {
     return async (req: Request, res: Response) => {
+      const { activeTab = 'placement' } = req.query as Record<string, PlacementTab>
```

Comment:
I'm not a fan of using a query string for the tab -- I'm not sure where the pattern was first introduced, but the routing framework lets us define paths easily, which can point to the same controller method and provide the `req.params` to determine what 'tab' should be shown, so we should be able to use clearly defined paths for each tab, e.g. `/manage/premises/:premisesId/placements/:placementId/application`, etc. At the moment, visiting `?activeTab=foo` does render the page but doesn't highlight any of the tabs as active. The routes also provide a way to define permissions, which can come in handy!

Reply:
True although the query string is used elsewhere. I'll refactor this to use explicit routes then.